orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/manifest/WaitForManifestStableTask.java

```diff
@@ -55,6 +57,20 @@ public long getTimeout() {
     return TimeUnit.MINUTES.toMillis(30);
   }
 
+  @Override
+  public long getDynamicTimeout(StageExecution stage) {
+    ManifestTimeout timeout = stage.mapTo(ManifestTimeout.class);
+    if (timeout.timeoutInMinutes != null) {
+      return TimeUnit.MINUTES.toMillis(timeout.timeoutInMinutes.get());
+    }
+
+    return getTimeout();
+  }
+
+  static class ManifestTimeout {
+    @JsonProperty Optional<Long> timeoutInMinutes;
```

Comment:
Does Jackson properly handle the `Optional` here?  As in, will it always set this to an empty `Optional` if the field is missing, and to an optional containing the field if it's present?  I've never seen this before, though would definitely believe it worked if there was a test on it 😁 

Also, as a really minor point, I think generally we omit the `in` from fields like this, so it would be `timeoutMinutes`.  (Either would in general be fine, but I think consistency with other fields would be good here.)

Reply:
I'm not sure on the Jackson side of things, but looking at other examples that's what they have: https://github.com/spinnaker/orca/blob/062e60fdea15e1ce75316535edfef09e61caf370/orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/manifest/WaitForManifestStableContext.java#L43

And I'll go ahead and remove the `in` from the field